### DISCUSSION TITLE
Should v and displacements in the MatDiffusion Kernel behave similarly ?

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#23666] [Q&A General] [answered]
Hi

I am trying to understand the difference between the `displacements` and the `v` in the `MatDiffusion` kernel [here](https://mooseframework.inl.gov/source/kernels/MatDiffusion.html) (and other `Diffusion` kernels as well). What I initially understood is that both of them are `couplable_variables`, am I right? If that is the case, why are they behaving differently? Please see below the input file that can reproduce my conclusion (replace `v` by `displacements` and the results will not be the same)

Another issue is w.r.t the jacobian computation when explicitly specifying `v`=`variable` (the default nonlinear variable).  As indicated in the documentation, if the `v` is not specified, the default nonlinear `variable` will be used for `v` (i.e. `v`=`variable`). Now, if the two lines in the kernel block were uncommented, it will result in an incorrect solution. However, if the `ADMatDiffusion` is used, it results in the correct solution, indicating the jacobian is not computed properly when explicitly specifying `v`=`variable`. 

I can work on a fix for these and submit a PR after confirming it and if you think it is worth fixing it.

Please see the expected (correct solution below):
![image](https://user-images.githubusercontent.com/88537645/223878543-86f0a158-187a-4f96-bab9-99c2f59e1481.png)

```
[Mesh]
  type = GeneratedMesh
  dim = 1
  xmin = 0
  xmax = 1
  nx = 1000
[]

[Variables]
  [ui]
  []
  [ur]
  []
[]

[BCs]
  [uxr_left_1]
    type = CoupledVarNeumannBC
    variable = ur
    boundary = 'left'
    v = ui
    coef = 99.627 
  []
  [uxr_left_2]
    type = CoupledVarNeumannBC
    variable = ur
    boundary = 'left'
    v = ur
    coef = -4.969 
  []

  [uxi_left_1]
    type = CoupledVarNeumannBC
    variable = ui
    boundary = 'left'
    v = ui
    coef = -4.969
  []
  [uxi_left_2]
    type = CoupledVarNeumannBC
    variable = ui
    boundary = 'left'
    v = ur
    coef = -99.627 
  []
  [right_real]
    type = DirichletBC
    variable = ur
    value = 1
    boundary = right
  []
  [right_imaginery]
    type = DirichletBC
    variable = ui
    value = 0
    boundary = right
  []
[]

[Kernels]
  [sigma_rr]
    type = MatDiffusion
    variable = ur
    diffusivity = 1
    # v = ur
  []
  [sigma_ii]
    type = MatDiffusion
    variable = ur
    diffusivity = -0.1
    v = ui
  []
  [sigma_ri]
    type = MatDiffusion
    variable = ui
    diffusivity = 1
    # v = ui
  []
  [sigma_ir]
    type = MatDiffusion
    variable = ui
    diffusivity = 0.1
    v = ur
  []
  [reaction_real]
    type = Reaction
    variable = ur
    rate = -10000
  []
  [reaction_imag]
    type = Reaction
    variable = ui
    rate = -10000
  []
[]

[Executioner]
  type = Steady
  solve_type = LINEAR
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
[]

[Outputs]
  exodus = true
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

Displacements is used to represent the deformation of the mesh, not a field living on the mesh.
Displacements affect the volume of elements for example. It s more involved than just solving for a field

Please submit a patch if you have found a bug. We can examine it in the PR

Guillaume
